src/storage/store/node_column.cpp

```diff
@@ -143,7 +148,9 @@ void BoolNodeColumn::batchLookup(
 
 void NodeColumn::scan(
     Transaction* transaction, ValueVector* nodeIDVector, ValueVector* resultVector) {
-    nullColumn->scan(transaction, nodeIDVector, resultVector);
+    if (propertyStatistics.mayHaveNull()) {
+        nullColumn->scan(transaction, nodeIDVector, resultVector);
+    }
```

Comment:
Are we assuming the `resultVector` will always be default to no nulls?
I don't think we should assume that. Instead, we should implicitly set `resultVector`'s null mask to all non null if the property statistics indicates not have nulls.

Reply:
Done. 

For both lookup and the scan variant with start and end positions I had it only set the specified parts of the result vector to non-null. Since in this case we know that no part of it is null, it would be safe to set the whole thing, but it might be faster to just set the specific locations, though I do wonder how much, particularly for random-access lookups where setNull could write multiple times to the same word.